redhat-appstudio/service-provider-integration-operator
# cleanup the token if it does not have any data

### What does this PR do?
If token data disappears from token storage for some reason, this PR ensures that both `SPIAccessToken` and `SPIAccesstokenBinding` are flipped back to `AwaitingTokenData` state. The injected Secret itself is also removed.


### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-413
https://issues.redhat.com/browse/SVPI-414

related PR: https://github.com/redhat-appstudio/service-provider-integration-operator/pull/512


### How to test this PR?


```
quay.io/redhat-appstudio/pull-request-builds:spi-controller-514
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-514
```

1. create SPIAccessTokenBinding
2. upload the token
3. delete data from token storage (e.g. connect to vault pod terminal and delete secret with `vault kv delete <path>`)
4. restart operator pod
5. check that `SPIAccessTokenBinding` and `SPIAccessToken` are in `AwaitingTokenData` and linked Secret is gone
